instrumentation/twilio-6.6/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/twilio/TwilioClientTest.java

```diff
@@ -0,0 +1,546 @@
+                            equalTo(stringKey("twilio.status"), "sent"))));
+  }
+
+  @Test
+  void synchronousCall() throws URISyntaxException {
+    twilioRestClient = mock(TwilioRestClient.class);
+    when(twilioRestClient.getObjectMapper()).thenReturn(new ObjectMapper());
+    when(twilioRestClient.request(any()))
+        .thenReturn(
+            new Response(
+                new ByteArrayInputStream(CALL_RESPONSE_BODY.getBytes(StandardCharsets.UTF_8)),
+                200));
+
+    Call call =
+        testing.runWithSpan(
+            "test",
+            () ->
+                Call.creator(
+                        new PhoneNumber("+15558881234"),
+                        new PhoneNumber("+15559994321"),
+                        new URI("http://twimlets.com/holdmusic?Bucket=com.twilio.music.ambient"))
+                    .create(twilioRestClient));
+
+    assertThat(call.getStatus()).isEqualTo(Call.Status.COMPLETED);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("test").hasNoParent().hasAttributes(Attributes.empty()),
+                span ->
+                    span.hasName("CallCreator.create")
+                        .hasKind(CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                stringKey("twilio.type"), "com.twilio.rest.api.v2010.account.Call"),
+                            equalTo(
+                                stringKey("twilio.account"), "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"),
+                            equalTo(stringKey("twilio.sid"), "CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"),
+                            equalTo(stringKey("twilio.status"), "completed"))));
+  }
+
+  @Test
+  void httpClient() throws IOException {
+    CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+    CloseableHttpResponse response1 = mockResponse(MESSAGE_RESPONSE_BODY, 200);
+    when(httpClient.execute(any())).thenReturn(response1);
+
+    HttpClientBuilder clientBuilder = getHttpClientBuilder(httpClient);
+
+    NetworkHttpClient networkHttpClient = new NetworkHttpClient(clientBuilder);
+
+    TwilioRestClient realTwilioRestClient =
+        new TwilioRestClient.Builder("username", "password")
+            .accountSid(ACCOUNT_SID)
+            .httpClient(networkHttpClient)
+            .build();
+
+    Message message =
+        testing.runWithSpan(
+            "test",
+            () ->
+                Message.creator(
+                        new PhoneNumber("+1 555 720 5913"),
+                        new PhoneNumber("+1 555 555 5215"),
+                        "Hello world!")
+                    .create(realTwilioRestClient));
+
+    assertThat(message.getBody()).isEqualTo("Hello, World!");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
```

Comment:
groovy test has an extra `POST` span

Reply:
I guess this happens because we can't instrument the mockito mock of `CloseableHttpClient` for some reason, but we could instrument the spock mock. If we really want to get the `POST` span we could create a fake http client by extending the `CloseableHttpClient` and implement the `doExecute` method to return the mocked response, though it is a bit of hassle. I think not having the `POST` spans is fine.

Reply:
Yes, Just as you thought ！ I take some time to find the `missing` span.  At last,  I chose not having the `POST` span , keep it clean.

Reply:
not having the POST span